### DISCUSSION TITLE
Kiali pod restarts frequently

### DISCUSSION THREAD

**Discussion by Yajnalatha** [kiali/kiali#5897] [Q&A]
Kiali pod restarts frequently and graph also doesn't load giving following error in pod log

1 portforward.go:391] error copying from local connection to remote stream: read tcp6 [::1]:14060->[::1]:40474: read: connection reset by peer

2023-03-07T08:54:56Z WRN GetPodProxyStatus is failing for [namespace: *****] [pod: ******]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port
E0307 08:54:59.564875       1 portforward.go:368] error creating forwarding stream for port 14070 -> 8080: EOF

**Comment by jmazzitelli** [comment]
_converted to a Q&A discussion - not clear that this is a bug_

You did not specify what version of Kiali and what version of Istio you are using and what cluster environment you are in (Google cloud? AWS? on-premise?). This is important to know for this kind of issue.

This istiod issue you are seeing has been seen in the past. Please search the Kiali github repo for the "unable to proxy istiod" and read the related issues and discussions. It may help.

IIRC, this is due to the fact you are in a cluster environment that does not expose istiod. I believe if you read prior discussions you should get more details.

**Reply by jmazzitelli** [reply]
Some previous discussions:
* https://github.com/kiali/kiali/discussions/4239
* https://github.com/kiali/kiali/discussions/4513
* https://github.com/kiali/kiali/issues/4679
* https://github.com/kiali/kiali/pull/5642
...

Here's a search query you can use to look up additional discussions on this: https://github.com/kiali/kiali/search?q=%22unable+to+proxy+istiod%22

**Comment by Yajnalatha** [comment]
I am using kiali 1.50.0, istio 1.15.3 and on-prem cluster

**Reply by jmazzitelli** [reply]
> kiali 1.50.0, istio 1.15.3 and on-prem cluster

https://kiali.io/docs/installation/installation-guide/prerequisites/#version-compatibility

![image](https://user-images.githubusercontent.com/2029470/223698987-e37bddb9-d0ac-470b-a814-3544017d04d1.png)

**Reply by jmazzitelli** [reply]
Your Kiali is incompatible with your Istio environment.
